core/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -397,6 +403,22 @@ RexNode isFalse(RexNode e) {
         : rexBuilder.makeCall(SqlStdOperatorTable.IS_FALSE, e);
   }
 
+  /**
+   * Runs simplification unary function by eliminating idempotent.
+   */
+  private RexCall simplifUnaryFunction(RexCall rexCall) {
```

Comment:
Maybe we can simplify `lower(upper(lower(x)))` to `upper(lower(x))` or `lower(upper(x))`?

Reply:
This is a bit like a special case, because the two functions have opposite effects. Maybe can hand this case, and it will not affect the default behavior.

Reply:
Yes，currently this goal can be achieved based on this PR. For example, lower(lower(upper(uper('a')))) will be simplified to lower(upper('a')).

Reply:
As @julianhyde  provided in the case of Jira, these two functions do not have opposite effects, so this case is not need to be simplified.